rotkehlchen/chain/ethereum/modules/convex/convex.py

```diff
@@ -0,0 +1,187 @@
+                    abi=pool.abi,
+                    method_name='balanceOf',
+                    arguments=[address],
+                )
+            except (BlockchainQueryError, RemoteError) as e:
+                log.debug(f'Failed to query convex balance of address {address} for pool {pool.address} due to {str(e)}')  # noqa: E501
+                continue
+            if lp_token_raw_amount == 0:
+                continue
+            lp_token_amount = token_normalized_value_decimals(
+                token_amount=lp_token_raw_amount,
+                token_decimals=18,
+            )
+            pool_tokens_base_amount = Inquirer().find_curve_pool_tokens_amount(
+                lp_token_address=lp_token_address,
+            )
+            if pool_tokens_base_amount is None:
+                continue
+            for token, (base_amount, usd_price) in pool_tokens_base_amount.items():
+                token_amount = base_amount * lp_token_amount
+                staking_balances.append(
+                    AssetBalance(
+                        asset=token,
+                        balance=Balance(
+                            amount=token_amount,
+                            usd_value=token_amount * usd_price,
+                        ),
+                    ),
+                )
+        return staking_balances
+
+    def get_history(
+            self,
+            addresses: List[ChecksumEthAddress],
+            reset_db_data: bool,  # pylint: disable=unused-argument
+            from_timestamp: Timestamp,
+            to_timestamp: Timestamp,
+    ) -> List[DefiEvent]:
+        db_events = DBHistoryEvents(database=self.database)
+        with db_events.db.conn.read_ctx() as cursor:
+            events_from_db = db_events.get_history_events(
+                cursor, HistoryEventFilterQuery.make(
+                    protocols=[CPT_CONVEX],
+                    location_labels=[str(addr) for addr in addresses],
+                    from_ts=from_timestamp,
+                    to_ts=to_timestamp,
+                ),
+                has_premium=self.premium is not None,
+            )
+        defi_events = []
+        for event in events_from_db:
+            pnl = got_asset = got_balance = spent_asset = spent_balance = None
+            if event.event_subtype == HistoryEventSubType.REWARD:
+                count_spent_got_cost_basis = True
+                pnl = [AssetBalance(asset=event.asset, balance=event.balance)]
+            else:
+                count_spent_got_cost_basis = False  # Staked assets remain yours
+            if event.event_type in (HistoryEventType.RECEIVE, HistoryEventType.WITHDRAWAL):
+                got_asset = event.asset
+                got_balance = event.balance
+            else:
+                spent_asset = event.asset
+                spent_balance = event.balance
+            defi_events.append(DefiEvent(
+                timestamp=Timestamp(event.timestamp * 1000),
```

Comment:
```suggestion
                timestamp=ts_sec_to_ms(event.timestamp),
```

btw question why are you increasing the precision here? In the databas is stored in ms right?